### PR TITLE
Add FFI bindings for MVP wallet features and generate test data

### PR DESCRIPTION
**Pull request by philipr-za**
## Description

This PR incorporates the beginning of the FFI API for the MVP wallet functionality. It also includes the means to generate rich test data in the wallet.

## Motivation and Context
Start of the FFI API for the wallet.

## How Has This Been Tested?
tests provided

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [x] New Tests
* [ ] Documentation

## Checklist:

* [x] I'm merging against the `development` branch
* [x] I ran `cargo-fmt --all` before pushing
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.

### CODE REVIEW

File: base_layer/wallet/src/test_utils.rs [hunk 1]
```diff
@@ -0,0 +1,253 @@
+// Copyright 2019. The Tari Project
```
Comment (CjS77) [inline base_layer/wallet/src/test_utils.rs]: call this testnet_utils? And tTo make it clear that it's not test code, please elaborate on the mod comment below

File: base_layer/wallet_ffi/cbindgen.toml [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+
+
+[export.rename]
+
+
+
+[export.body]
+
+
```
Comment (CjS77) [inline base_layer/wallet_ffi/cbindgen.toml]: A looot of whitespace

File: base_layer/wallet_ffi/wallet.h [hunk 3]
```diff
@@ -0,0 +1,130 @@
+//  wallet_ffi.h
```
Comment (CjS77) [inline base_layer/wallet_ffi/wallet.h]: Can this be called libwallet at all?

File: base_layer/wallet/src/test_utils.rs (hunk 1 above)
Comment (philipr-za) [reply base_layer/wallet/src/test_utils.rs]: Changed the name and added a more explicit comment at the top of the file. In my next PR I hide those whole file behind a feature flag to make it even clearer.

File: base_layer/wallet_ffi/cbindgen.toml (hunk 2 above)
Comment (philipr-za) [reply base_layer/wallet_ffi/cbindgen.toml]: removed

File: base_layer/wallet_ffi/wallet.h (hunk 3 above)
Comment (philipr-za) [reply base_layer/wallet_ffi/wallet.h]: sure

**Pull request merged**